BuildScript.UnitTests/FilterTestConfigurationsTest.cs

```diff
@@ -0,0 +1,498 @@
+      var testConfigurations = new[] { testConfiguration };
+      var filter = CreateFilterTestConfigurations (supportedBrowsers: new string[0]);
+
+      var filteredTestConfigurations = filter.GetFilteredTestConfigurations (testConfigurations);
+
+      Assert.That (filteredTestConfigurations, Is.EqualTo (testConfigurations));
+    }
+
+    [Test]
+    public void NoDBAndEmptySupportedDatabaseSystems_IgnoresCase ()
+    {
+      var testConfiguration = CreateTestConfiguration (databaseSystem: "nodb");
+      var testConfigurations = new[] { testConfiguration };
+      var filter = CreateFilterTestConfigurations (supportedDatabaseSystems: new string[0]);
+
+      var filteredTestConfigurations = filter.GetFilteredTestConfigurations (testConfigurations);
+
+      Assert.That (filteredTestConfigurations, Is.EqualTo (testConfigurations));
+    }
+
+    [Test]
+    public void SupportedDatabaseSystems_NoDBIsAlwaysSupported ()
+    {
+      var testConfiguration = CreateTestConfiguration (databaseSystem: "NoDB");
+      var testConfigurations = new[] { testConfiguration };
+      var filter = CreateFilterTestConfigurations();
+
+      Assert.That (() => filter.GetFilteredTestConfigurations (testConfigurations), Throws.Nothing);
+    }
+
+    [Test]
+    public void SupportedExecutionRuntimes_LocalMachineIsAlwaysSupported ()
+    {
+      var testConfiguration = CreateTestConfiguration (executionRuntime: "LocalMachine");
+      var testConfigurations = new[] { testConfiguration };
+      var allExecutionRuntimes = new[] { "LocalMachine" };
+      var filter = CreateFilterTestConfigurations (enabledExecutionRuntimes: allExecutionRuntimes);
+
+      Assert.That (() => filter.GetFilteredTestConfigurations (testConfigurations), Throws.Nothing);
+    }
+
+    [Test]
+    public void SupportedBrowsers_NoBrowserIsAlwaysSupported ()
+    {
+      var testConfiguration = CreateTestConfiguration (browser: "NoBrowser");
+      var testConfigurations = new[] { testConfiguration };
+      var filter = CreateFilterTestConfigurations();
+
+      Assert.That (() => filter.GetFilteredTestConfigurations (testConfigurations), Throws.Nothing);
+    }
+
+    [Test]
+    public void SupportedBrowsers_FilterForUnsupportedBrowser_Error ()
+    {
+      const string unsupportedBrowser = "SomeUnsupportedBrowser";
+      var testConfiguration = CreateTestConfiguration (browser: unsupportedBrowser);
+      var testConfigurations = new[] { testConfiguration };
+      var supportedBrowsers = new[] { "Chrome" };
+      var loggerMock = MockRepository.GenerateMock<ITaskLogger>();
+      loggerMock.Expect (_ => _.LogError ($"Cannot filter with unsupported {TestingConfigurationMetadata.Browser} '{unsupportedBrowser}'"));
+      var filter = CreateFilterTestConfigurations (supportedBrowsers: supportedBrowsers, logger: loggerMock);
+
+      Assert.That (() => filter.GetFilteredTestConfigurations (testConfigurations), Throws.Exception);
```

Comment:
I left it out since the exception type and message are not important, the log message is. I wouldn't assert the message at this point, but I could assert the type